libvast/vast/system/component_registry.hpp

```diff
@@ -79,6 +82,37 @@ class component_registry {
     return {find_by_label(l0), find_by_label(l1), find_by_label(ls)...};
   }
 
+  /// Locates typed components by handle.
+  /// @tparam Handles... The typed actor handles to lookup.
+  /// @returns The respective component actors if found.
+  template <class... Handles>
+  std::tuple<Handles...> find() const {
+    auto normalize = [](std::string in) {
+      // Remove the uninteresting parts of the name:
+      //   vast::system::type_registry_actor -> type_registry
+      in.erase(0, sizeof("vast::system::") - 1);
```

Comment:
This computation requires some assertions, e.g., `starts_with(str, "vast::system::")`.

Unless the registration with labe also works as dual to this transformation, I don't feel comfortable with this method. @tobim?

Reply:
Note that this transformation code was written by @tobim originally. Let's talk about this tomorrow briefly. I think it's fine as is for now and we can change it up when making the node itself a typed actor.

Reply:
Good to know. I think it's actually not complicated to switch from

```cpp
const char* singletons[]
    = {"accountant", "archive", "eraser",       "filesystem",
       "importer",   "index",   "type-registry"};
```

to a type list in the node. My main worry is that these go out of sync and then we're in trouble.

I agree it's best to chat about this briefly.

Reply:
I agree with the general sentiment that these should be generated from the same expressions or checked statically. 

> My main worry is that these go out of sync and then we're in trouble.

When one changes one but not the other during development we'll see some failing unit or integration tests. That won't give a good indication of what is wrong, and makes the dev waste time on debugging.

Reply:
@dominiklohmann is it feasible to generate `singletons` from the type ids and centralize (or drop?) the normalization?

Reply:
Let's do this only after we converted the node to a typed actor interface. It's almost impossible to catch all possible issues with this currently.

Reply:
I added a comment above `singletons` that explains what we should do here in the future.

Reply:
@tobim does my comment suffice? Can you re-review?

Reply:
> Let's do this only after we converted the node to a typed actor interface.

Alright, fine with me. Can you take care of creating a corresponding story?

Reply:
Will do before merging this.

Reply:
Two stories exist now as a follow-up. @tobim / @mavam mind approving (unless you want anything else changed)?